solidity/test/BondedECDSAKeepFactoryTest.js

```diff
@@ -1046,6 +1046,65 @@ contract("BondedECDSAKeepFactory", async accounts => {
             )
         })
 
+        it("records the keep address and opening time", async () => {
+            let preKeepCount = await keepFactory.getKeepCount()
+
+            let keepAddress = await keepFactory.openKeep.call(
+                groupSize,
+                threshold,
+                keepOwner,
+                bond,
+                { from: application, value: feeEstimate }
+            )
+
+            await keepFactory.openKeep(
+                groupSize,
+                threshold,
+                keepOwner,
+                bond,
+                { from: application, value: feeEstimate }
+            )
+            let recordedKeepAddress = await keepFactory.getKeepAtIndex(preKeepCount)
+            let keep = await BondedECDSAKeep.at(keepAddress)
+            let keepCreationTime = await keep.getTimestamp()
+            let factoryCreationTime = await keepFactory.getCreationTime(keepAddress)
+
+            assert.equal(
+                recordedKeepAddress,
+                keepAddress,
+                "address recorded in factory differs from returned keep address",
+            );
+
+            expect(
+                factoryCreationTime
+            ).to.eq.BN(
+                keepCreationTime,
+                "creation time in factory differs from creation time in keep",
+            )
+        })
+
+        it("produces active keeps", async () => {
+            let keepAddress = await keepFactory.openKeep.call(
+                groupSize,
+                threshold,
+                keepOwner,
+                bond,
+                { from: application, value: feeEstimate }
+            )
+
+            await keepFactory.openKeep(
+                groupSize,
+                threshold,
+                keepOwner,
+                bond,
+                { from: application, value: feeEstimate }
+            )     
+
+            let keep = await BondedECDSAKeep.at(keepAddress)
+
+            assert.isTrue(await keep.isActive(), "keep should be active")
+        })
+
```

Comment:
Can we add test cases for `getKeepAtIndex` where `index = keeps.length` and `index > keeps.length` to check if it reverts?

Reply:
See https://github.com/keep-network/keep-ecdsa/pull/333/commits/823124cbf42af77886bcc21cc676fa14b36d0243